spec/services/import_polygons_spec.rb

```diff
@@ -2,84 +2,151 @@
+          expect(LocationPolygon.counties.count).to eq(1)
+        end
 
-      it "imports Tower Hamlets local authority" do
-        expect(LocationPolygon.london_boroughs.first.name).to eq("tower hamlets")
+        it "imports Essex boundaries" do
+          expect(LocationPolygon.counties.first.boundary).to eq(boundary_points)
+        end
       end
 
-      it "does not import non-existent local authority" do
-        expect(LocationPolygon.london_boroughs.count).to eq(1)
+      context "when location type is london boroughs" do
+        let(:location_type) { :london_boroughs }
+        let(:boundary_points) { [51.5083356630733, 0.00456249603, 51.5051084208278, -0.0057105307948] }
+        let(:file_name) { "ons_london_boroughs.json" }
+
+        it "imports Tower Hamlets local authority" do
+          expect(LocationPolygon.london_boroughs.first.name).to eq("tower hamlets")
+        end
+
+        it "does not import non-existent local authority" do
+          expect(LocationPolygon.london_boroughs.count).to eq(1)
+        end
+
+        it "imports Tower Hamlets boundaries" do
+          expect(LocationPolygon.london_boroughs.first.boundary).to eq(boundary_points)
+        end
       end
 
-      it "imports Tower Hamlets boundaries" do
-        expect(LocationPolygon.london_boroughs.first.boundary).to eq(points)
+      context "when location type is cities" do
+        let(:location_type) { :cities }
+        let(:boundary_points) { [51.406361958644, -2.3780576677997, 51.4063596372237, -2.3787764623145] }
+        let(:file_name) { "ons_cities.json" }
+
+        it "imports Bath city" do
+          expect(LocationPolygon.cities.first.name).to eq("bath")
+        end
+
+        it "does not import non-existent city" do
+          expect(LocationPolygon.cities.count).to eq(1)
+        end
+
+        it "imports Bath boundaries" do
+          expect(LocationPolygon.cities.first.boundary).to eq(boundary_points)
+        end
       end
     end
 
-    context "when location type is cities" do
-      let(:location_type) { :cities }
-      let(:points) { [51.406361958644, -2.3780576677997, 51.4063596372237, -2.3787764623145] }
-      let(:file_name) { "ons_cities.json" }
-
-      it "imports Bath city" do
-        expect(LocationPolygon.cities.first.name).to eq("bath")
+    describe "#get_buffers" do
+      let(:buffer_response) { JSON.parse(file_fixture("buffer_response.json").read) }
+      let(:buffer_coordinates) { [65.06414131400004, -12.5761721229999353, 65.06185937400005, -12.5761901179999427] }
+      let(:imported_buffers) do
+        {
+          "5" => buffer_coordinates,
+          "10" => buffer_coordinates,
+          "15" => buffer_coordinates,
+          "20" => buffer_coordinates,
+          "25" => buffer_coordinates,
+        }
       end
 
-      it "does not import non-existent city" do
-        expect(LocationPolygon.cities.count).to eq(1)
+      context "when api_endpoint is shorter than URL_MAXIMUM_LENGTH" do
```

Comment:
Let's make the two context descriptions more informative to explain why the api_endpoint length could vary. We should also remove references to implementation details such as the variable name and the constant name.

E.g. "when the length of the params caused the API endpoint length to exceed the maximum"